### ISSUE TITLE
searchApp.js startup should be delayed

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#38235]
The Search process should be delayed until after the workbench is completely initialized (explorer restored, editors restored, ext host up and running):

The dark red shows that the `searchApp.js` competes for OS resources with the ext host process (purple). I can share the tool that generates this when I polish it a bit:
![image](https://user-images.githubusercontent.com/5047891/32738417-1e35de24-c89d-11e7-8f87-484ed2371c8f.png)

**Comment by alexandrudima** [comment]
Adding @bpasero to provide guidance what would be a good workbench event to listen to and begin warming up/starting up the search service.

**Comment by roblourens** [comment]
The search process only starts on request, so when it starts that early, it's probably due to extensions with `workspaceContains` patterns. I agree that we should delay it.

**Comment by alexandrudima** [comment]
@roblourens This is not what profiling via ProcMon or logging shows. It shows that the search service is always spawned eagerly on startup:

Here is a change I do. I add `console.log(`forking ${this.modulePath}, ${JSON.stringify(args)}`);` after the line at 
https://github.com/Microsoft/vscode/blob/fd49805388c1d956eefdcb610956bbfed12db6db/src/vs/base/parts/ipc/node/ipc.cp.ts#L148

I see the following console output:
![image](https://user-images.githubusercontent.com/5047891/32739401-0d4412ea-c8a0-11e7-8ff3-340bc940f7ee.png)

Can you please double check that the search process only starts on request ?

**Comment by roblourens** [comment]
Actually since this change https://github.com/Microsoft/vscode/commit/c354fca44896da2c9cad82b9bd9126f6658cfae8 the search process is forced to started on launch. If I remove the `forwardTelemetry()` call in the SearchService constructor, then it doesn't start the search service until the user does a search.

@chrmarti Can that call be delayed until a user's search request?

**Comment by chrmarti** [comment]
Yes, my issue.

**Comment by bpasero** [comment]
@roblourens @chrmarti you can see [here](https://github.com/Microsoft/vscode/commit/994ffd6890a4d17ae3af6da868c1b1c9e7d11e97) how I delay starting of the file watcher until we are running. 

The idea is that the lifecycle service has phases that you can join on:
* `LifecyclePhase.Starting`: right on startup (earliest)
* `LifecyclePhase.Restoring`: right before we start to restore views and editors
* `LifecyclePhase.Running`: after we have restored views and editors
* `LifecyclePhase.RunningForABit`: 3 seconds after we have restored views and editors

@alexandrudima did you see the file watcher spawning also show up?

**Comment by chrmarti** [comment]
For the search app we can postpone the launch to the first text search / QuickOpen. It used to be like that until I inadvertently changed it.

**Comment by roblourens** [comment]
I think that it will still be activated on launch by extensions with `workspaceContains` patterns, so we can also either force it to wait in that case too, or allow it in that case.

**Comment by dbaeumer** [comment]
Verified that now search service is started when code starts.
